src/Utilities/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -435,7 +435,7 @@ private bool IsMethodArgumentASink(IMethodSymbol method, IEnumerable<IArgumentOp
                 Lazy<HashSet<SinkKind>> lazySinkKinds = new Lazy<HashSet<SinkKind>>(() => new HashSet<SinkKind>());
                 foreach (SinkInfo sinkInfo in this.DataFlowAnalysisContext.SinkInfos.GetInfosForType(method.ContainingType))
```

Comment:
suggestion: `var sinkInfos = this.DataFlowAnalysisContext.SinkInfos.GetInfosForType(method.ContainingType).Where(si => !(lazySinkKinds.IsValueCreated && lazySinkKinds.Value.IsSupersetOf(si));`  then `foreach (var sinkInfo in sinkInfos)`

Reply:
I think I'd find that hard to follow, and `lazySinkKinds` is mutable while iterating, so I don't think it belongs in `Where()`.